src/group_theory/quotient_group.lean

```diff
@@ -420,3 +420,38 @@ begin
 end
 
 end quotient_group
+
+namespace group
+
+open_locale classical
+open subgroup quotient_group
+
+variables {F G H : Type u} [group F] [group G] [group H] [fintype F] [fintype H]
+variables {f : F →* G} {g : G →* H}
+
+/-- If `F` and `H` are finite such that `ker(G →* H) ≤ im(F →* G)`, then `G` is finite. -/
+@[to_additive "If `F` and `H` are finite such that `ker(G →+ H) ≤ im(F →+ G)`, then `G` is finite."]
+noncomputable def fintype_of_ker_le_range (h : g.ker ≤ f.range) : fintype G :=
+@fintype.of_equiv _ _
+  (@prod.fintype _ _ (fintype.of_injective _ $ ker_lift_injective g) $
+                      fintype.of_injective _ $ inclusion_injective h)
+  group_equiv_quotient_times_subgroup.symm
+
+/-- If `F` and `H` are finite such that `ker(G →* H) = im(F →* G)`, then `G` is finite. -/
+@[to_additive "If `F` and `H` are finite such that `ker(G →+ H) = im(F →+ G)`, then `G` is finite."]
+noncomputable def fintype_of_ker_eq_range (h : g.ker = f.range) : fintype G :=
+fintype_of_ker_le_range (eq_iff_le_not_lt.mp h).left
+
+/-- If `ker(G →* H)` and `H` are finite, then `G` is finite. -/
+@[to_additive "If `ker(G →+ H)` and `H` are finite, then `G` is finite."]
+noncomputable def fintype_of_ker_of_codom [fintype g.ker] : fintype G :=
+@fintype_of_ker_le_range g.ker _ _ _ _ _ _ _
```

Comment:
I think this `@fintype_of_ker_le_range` (and the one below) is an indicator that you should make a few more variables explicit.